src/server/game/Entities/Creature/Creature.h

```diff
@@ -448,7 +448,7 @@ class Creature : public Unit, public GridObject<Creature>, public MapObject
         bool IsTrigger() const { return (GetCreatureTemplate()->flags_extra & CREATURE_FLAG_EXTRA_TRIGGER) != 0; }
         bool IsGuard() const { return (GetCreatureTemplate()->flags_extra & CREATURE_FLAG_EXTRA_GUARD) != 0; }
         bool CanWalk() const { return (GetCreatureTemplate()->InhabitType & INHABIT_GROUND) != 0; }
-        bool CanSwim() const { return (GetCreatureTemplate()->InhabitType & INHABIT_WATER) != 0 || IsPet(); }
```

Comment:
explain? it was moved to .cpp - do you have any objections for new implementation?